core/src/main/scala/org/broadinstitute/dsde/rawls/workspace/WorkspaceService.scala

```diff
@@ -338,25 +340,39 @@ class WorkspaceService(protected val userInfo: UserInfo, val dataSource: SlickDa
     }
   }
 
-  def updateWorkspace(workspaceName: WorkspaceName, operations: Seq[AttributeUpdateOperation]): Future[PerRequestMessage] =
-    withAttributeNamespaceCheck(operations.map(_.name)) {
-      dataSource.inTransaction { dataAccess =>
-        withWorkspaceContextAndPermissions(workspaceName, WorkspaceAccessLevels.Write, dataAccess) { workspaceContext =>
-          val updateAction = Try {
-            val updatedWorkspace = applyOperationsToWorkspace(workspaceContext.workspace, operations)
-            dataAccess.workspaceQuery.save(updatedWorkspace)
-          } match {
-            case Success(result) => result
-            case Failure(e: AttributeUpdateOperationException) =>
-              DBIO.failed(new RawlsExceptionWithErrorReport(errorReport = ErrorReport(StatusCodes.BadRequest, s"Unable to update ${workspaceName}", ErrorReport(e))))
-            case Failure(regrets) => DBIO.failed(regrets)
-          }
-          updateAction.map { savedWorkspace =>
-            RequestComplete(StatusCodes.OK, savedWorkspace)
-          }
+  def updateLibraryAttributes(workspaceName: WorkspaceName, operations: Seq[AttributeUpdateOperation]): Future[PerRequestMessage] = {
+    withLibraryAttributeNamespaceCheck(operations.map(_.name)) {
+      getWorkspaceContext(workspaceName) flatMap { ctx =>
+        withLibraryPermissions(ctx.workspace, operations, dataSource, userInfo, getMaximumAccessLevel _) {
+          updateWorkspace(workspaceName, operations, ctx)
         }
       }
     }
+  }
+
+  def updateWorkspace(workspaceName: WorkspaceName, operations: Seq[AttributeUpdateOperation]): Future[PerRequestMessage] = {
+    withAttributeNamespaceCheck(operations.map(_.name)) {
+    getWorkspaceContextAndPermissions(workspaceName, WorkspaceAccessLevels.Write) flatMap { ctx =>
+        updateWorkspace(workspaceName, operations, ctx)
+      }
+    }
+  }
```

Comment:
nothing prevents a user from calling this with library attributes and bypassing the library security checks! Maybe update `attributeNamespaceCheck` to reject library namespace?